sympy/core/numbers.py

```diff
@@ -967,6 +967,8 @@ def __new__(cls, num, dps=None, prec=None, precision=None):
             num = '+inf'
         elif num is S.NegativeInfinity:
             num = '-inf'
+        elif type(num).__module__ == 'numpy': # support for numpy datatypes
+            num = convert_numpy_types(num)
```

Comment:
Here you're converting the numpy obj to a Sympy object and letting the rest of the `Float.__new__` deal with the other details.  This looks like the right idea if it works.

Reply:
Hmmm, what would happen if we just to use `num = sympify(num)` here?  No other changes needed...?

Reply:
I initially thought that ```num = sympify(num)``` would cause an endless recursion, but I see now that that's not the case. In fact, after making that change and running it through @Trave11er's tests, it seems to work fine. Incorporated into next commit.